internal/controlplane/common.go

```diff
@@ -35,6 +37,12 @@ type ProjectIDGetter interface {
 	GetProjectId() string
 }
 
+// ProviderNameGetter is an interface that can be implemented by a request
+type ProviderNameGetter interface {
+	// GetProvider returns the provider name
+	GetProvider() string
+}
+
```

Comment:
I ❤️  creating an interface like this to fetch the common field across many different protos.  Nice job!